### PR TITLE
[koa-static] upgrade to v4.0

### PR DESCRIPTION
**Pull request by tlaziuk**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/koajs/static/blob/master/History.md>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/koa-send/index.d.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ declare namespace send {
         /** Root directory to restrict file access. (defaults to '') */
         root?: string;
         /** Name of the index file to serve automatically when visiting the root location. (defaults to none) */
-        index?: string;
+        index?: string | false;
```
Comment (pe8ter) [inline types/koa-send/index.d.ts]: Why `false` here? I don't see how the behavior of koa-send would change if `index` were `false`.

Comment (tlaziuk) [reply types/koa-send/index.d.ts]: That's true as `''` (empty string) is a falsely value, and  for `koa-send` that is enough (https://github.com/koajs/send/blob/master/index.js#L66 https://github.com/koajs/send/blob/master/index.js#L111).

But `koa-static` uses comparison to `false` (https://github.com/koajs/static/blob/master/index.js#L36), so I'd like to have your advise in this case @pe8ter.
Maybe also other options (eg. `root` - https://github.com/koajs/send/blob/master/index.js#L43) should be able to accept falsely types (`void`, `false`, `number` - I'm not sure with this one).

Comment (aozgaa) [reply types/koa-send/index.d.ts]: Would it be possible to make this PR without this change and then re-open the question in a separate PR? Everything else looks reasonable to me.

Comment (paulvanbrenk) [reply types/koa-send/index.d.ts]: @pe8ter can reply to the question by @tlaziuk.
@tlaziuk can you create a separate PR for that change... so we can get the rest in?

Comment (pe8ter) [reply types/koa-send/index.d.ts]: I now disagree with my earlier objection. One should explicitly use `false` to disable the default _index.html_.

**Comment by pe8ter** [comment]
👍

**Pull request merged**